web/src/main/java/com/microsoft/applicationinsights/web/internal/correlation/CdsProfileFetcher.java

```diff
@@ -133,6 +152,15 @@ public void setEndpointAddress(String endpoint) throws MalformedURLException {
         return this.httpClient.execute(request, null);
     }
 
+    private synchronized void incrementFailureCount(String instrumentationKey) {
```

Comment:
We could possibly remove synchornization from this method if we make the failureCounters HashMap to be a thread safe concurrent HashMap because we only need atomicity in adding the values to the map.